aten/src/ATen/TensorIterator.cpp

```diff
@@ -396,6 +406,13 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
                   "desired output type ", op.current_dtype);
     }
 
+    // Checks safe casting for inputs if common_dtype was specified by user.
+    if (config.specified_common_dtype != ScalarType::Undefined && !op.is_output && op.current_dtype != common_dtype_) {
```

Comment:
I think this check can be moved into the above conditional, and doing so will probably reduce the instruction count.